src/tools/wix/ChainPackageInfo.cs

```diff
@@ -611,7 +611,23 @@ private void ResolveMsiPackage(BinderFileManager fileManager, Dictionary<string,
 
                     if (!Common.IsValidModuleOrBundleVersion(this.Version))
                     {
-                        this.core.OnMessage(WixErrors.InvalidProductVersion(this.PackagePayload.SourceLineNumbers, this.Version, sourcePath));
+                        // not a proper .NET version (i.e., five fields); can we get a valid three-part version number?
```

Comment:
I don't follow your comment about major upgrades.  Burn will consider 1.0.0.0 to 1.0.0.1 a major upgrade assuming a standard Upgrade table.